accepted/future-releases/0546-patterns/feature-specification.md

```diff
@@ -2852,9 +2852,20 @@ To match a pattern `p` against a value `v`:
 
     5.  Match the head elements. For `i` from `0` to `h - 1`, inclusive:
 
-        1.  Extract the element value `e` by calling `v[i]`.
+        1.  Let `s` be the `i`th element subpattern.
 
-        2.  Match the `i`th element subpattern against `e`.
+        2.  If `s` is an identifier pattern whose name is `_` then do nothing
+            for this element.
+
+            *Wildcards are useful in list patterns to control the index that
+            latter element subpatterns access and to affect the length that the
+            pattern checks. When a user does that, we don't want them to pay a
+            runtime penalty for accessing list elements that the pattern won't
+            use anyway.*
+
+        3.  Else extract the element value `e` by calling `v[i]`.
+
+        4.  Match `s` against `e`.
 
     6.  If there is a matching rest element:
```

Comment:
Should this be "If there is a rest element:"? We don't know if it's matching before we do the thing below.

Reply:
Earlier in the proposal, we define:

> We refer to a rest element with a subpattern as a *matching rest element*, and a rest element with no subpattern as a *non-matching rest element*.

So it just means "rest element with a subpattern", not necessarily "rest element whose subpattern has successfully matched."